lib/sync/fetcher.go

```diff
@@ -133,6 +135,16 @@ func (f *BlockFetcher) fetch(ctx context.Context, si *SyncInfo) error {
 
 	items, err := f.unmarshalResp(resp.Body)
 	if err != nil {
+		err := errors.Wrap(err, "reponse failed to unmarshal")
+
+		body := func() string {
+			body, readErr := ioutil.ReadAll(resp.Body)
```

Comment:
Won't we always get `EOF` here if the reader has been consumed in `unmarshalResp` ?

Reply:
I did use a buffer for it.  But I have a mind to remove body logging.. too.  :-|